base/test/valveModule_tests.cpp

```diff
@@ -0,0 +1,330 @@
+    {
+        frame_container sink2Frames = sink2Que->pop();
+        sink2Frames = sink2Que->pop();
+        BOOST_TEST(sink2Frames.size() == 2);
+        bool flagGen = false;
+        bool flagRaw = false;
+        for (auto framePair = sink2Frames.begin(); framePair != sink2Frames.end(); framePair++)
+        {
+            auto metadata = (framePair->second)->getMetadata();
+            if ((metadata->getFrameType()) == FrameMetadata::GENERAL)
+            {
+                flagGen = true;
+            }
+            if ((metadata->getFrameType()) == FrameMetadata::RAW_IMAGE)
+            {
+                flagRaw = true;
+            }
+        }
+        BOOST_TEST(flagGen);
+        BOOST_TEST(flagRaw);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(getSetProps)
+{
+    auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+    auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::GENERAL));
+    auto pinId = source->addOutputPin(metadata);
+    auto valve = boost::shared_ptr<ValveModule>(new ValveModule(ValveModuleProps(2)));
+    source->setNext(valve, true, false);
+    auto sink = boost::shared_ptr<SinkModule>(new SinkModule(SinkModuleProps()));
+    valve->addOutputPin(metadata);
+    valve->setNext(sink, true, false);
+
+
+    BOOST_TEST(source->init());
+    BOOST_TEST(valve->init());
+    BOOST_TEST(sink->init());
+
+    auto sinkQue = sink->getQue();
+
+    auto frame = source->makeFrame(1023, pinId);
+    frame_container frames;
+    frames.insert(make_pair(pinId, frame));
+
+    // default props is 2
+    valve->allowFrames(2);
+    valve->step();
+
+    // We are sending 4 frames with enable true so only two frames (moduleprops = 2) pass through the valve
+    for (int i = 0; i < 4; i++)
+    {
+        source->send(frames);
+        valve->step();
+    }
+
+    BOOST_TEST(sinkQue->size() == 2);
+
+    //We are resetting the moduleprops to 4 and enable true 
+    // rvw - getProps - change the value - sending in setProps
+    auto currentProps = valve->getProps();
+    currentProps.noOfFramesToCapture = 4;
+    auto newValue = ValveModuleProps(currentProps.noOfFramesToCapture);
+    valve->setProps(newValue);
+    valve->step();
+    valve->step();
+
+    for (int i = 0; i < 4; i++)
+    {
+        source->send(frames);
+        valve->step();
+    }
+ 
+    BOOST_TEST(sinkQue->size() == 6);
+
+    //We are sending 4 frames again with enable false
+    for (int i = 0; i < 4; i++)
+    {
+        source->send(frames);
+        valve->step();
+    }
+
+    BOOST_TEST(sinkQue->size() == 6);
+
+}
+
+BOOST_AUTO_TEST_CASE(open)
+{
+    auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+    auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::GENERAL));
+    auto pinId = source->addOutputPin(metadata);
+    auto valve = boost::shared_ptr<ValveModule>(new ValveModule(ValveModuleProps(-1)));
+    source->setNext(valve, true, false);
```

Comment:
same comment

Reply:
Also, this setNext should be overriden in the valveModule to allow `valve->setNext(sinkModule)`